sdk/lib/drivers/lwip/src/rostcp.c

```diff
@@ -647,11 +647,18 @@ LibTCPShutdownCallback(void *arg)
      * PCB without telling us if we shutdown TX and RX. To avoid these problems, we'll clear the
      * socket context if we have called shutdown for TX and RX.
      */
-    if (msg->Input.Shutdown.shut_rx) {
-        msg->Output.Shutdown.Error = tcp_shutdown(pcb, TRUE, FALSE);
+    if (msg->Input.Shutdown.shut_rx != msg->Input.Shutdown.shut_tx) {
```

Comment:
This this also hold up if both are `0` ?

Reply:
If both are 0, nothing should be done I think (and there should be a test case for this ^^)

Reply:
Yeah could be an option, this case shouldn't even happen